### PR TITLE
Convert controller deployment files to structured logs

### PR DESCRIPTION
**Pull request by scottilee**
#### What type of PR is this?
/kind cleanup
/sig instrumentation

#### What this PR does / why we need it:
Converts log statements to use structured logging.

#### Which issue(s) this PR fixes:
N/A

#### Special notes for your reviewer:
N/A

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs
- [KEP]: https://github.com/kubernetes/enhancements/blob/master/keps/sig-instrumentation/1602-structured-logging/kep.yaml
```

### CODE REVIEW

File: pkg/controller/deployment/util/deployment_util.go [hunk 1]
```diff
@@ -280,7 +280,7 @@ func SetNewReplicaSetAnnotations(deployment *apps.Deployment, newRS *apps.Replic
 				oldRevisions = append(oldRevisions[start:], oldRevision)
 				newRS.Annotations[RevisionHistoryAnnotation] = strings.Join(oldRevisions, ",")
 			} else {
-				klog.Warningf("Not appending revision due to length limit of %v reached", revHistoryLimitInChars)
+				klog.Warningf("Not appending revision due to length limit being reached", "lengthLimit", revHistoryLimitInChars)
```
Comment (yangjunmyfm192085) [inline pkg/controller/deployment/util/deployment_util.go:283]: Maybe InfoS?

Comment (scottilee) [reply pkg/controller/deployment/util/deployment_util.go:283]: Fixed.

Comment (scottilee) [reply pkg/controller/deployment/util/deployment_util.go]: Fixed.

**Comment by scottilee** [comment]
@yangjunmyfm192085 Addressed your comment. Let me know if I missed anything else.

**Comment by scottilee** [comment]
Addressed failures in "pull-kubernetes-verify" test.

File: pkg/controller/deployment/rolling.go [hunk 2]
```diff
@@ -91,7 +91,7 @@ func (dc *DeploymentController) reconcileOldReplicaSets(allRSs []*apps.ReplicaSe
 	}
 
 	allPodsCount := deploymentutil.GetReplicaCountForReplicaSets(allRSs)
-	klog.V(4).Infof("New replica set %s/%s has %d available pods.", newRS.Namespace, newRS.Name, newRS.Status.AvailableReplicas)
+	klog.V(4).InfoS("New ReplicaSet has available Pods.", "replicaset", klog.KObj(newRS), "AvailableReplicas", newRS.Status.AvailableReplicas)
```
Comment (yangjunmyfm192085) [inline pkg/controller/deployment/rolling.go:94]:
```suggestion
	klog.V(4).InfoS("New ReplicaSet has available Pods", "replicaset", klog.KObj(newRS), "AvailableReplicas", newRS.Status.AvailableReplicas)
```

File: pkg/controller/deployment/util/deployment_util.go [hunk 3]
```diff
@@ -203,7 +203,7 @@ func LastRevision(allRSs []*apps.ReplicaSet) int64 {
 	for _, rs := range allRSs {
 		if v, err := Revision(rs); err != nil {
 			// Skip the replica sets when it failed to parse their revision information
-			klog.V(4).Infof("Error: %v. Couldn't parse revision for replica set %#v, deployment controller will skip it when reconciling revisions.", err, rs)
+			klog.V(4).InfoS("Couldn't parse revision for ReplicaSet, deployment controller will skip it when reconciling revisions.", "replicaset", klog.KObj(rs), "err", err)
```
Comment (yangjunmyfm192085) [inline pkg/controller/deployment/util/deployment_util.go:206]:
```suggestion
			klog.V(4).InfoS("Couldn't parse revision for ReplicaSet, deployment controller will skip it when reconciling revisions", "replicaset", klog.KObj(rs), "err", err)
```

File: pkg/controller/deployment/rolling.go (hunk 2 above)
Comment (scottilee) [reply pkg/controller/deployment/rolling.go:94]: Fixed.

File: pkg/controller/deployment/util/deployment_util.go (hunk 3 above)
Comment (scottilee) [reply pkg/controller/deployment/util/deployment_util.go:206]: Fixed.

**Comment by scottilee** [comment]
@yangjunmyfm192085 I addressed your comments.

**Pull request closed without merging**